src/components/tl/ucp/allreduce/allreduce_knomial.c

```diff
@@ -78,8 +78,7 @@ ucc_status_t ucc_tl_ucp_allreduce_knomial_progress(ucc_coll_task_t *coll_task)
                              (status = ucc_dt_reduce(sbuf, scratch, rbuf, count,
                                                      dt, mem_type, args)))) {
                 tl_error(UCC_TASK_LIB(task), "failed to perform dt reduction");
-                task->super.super.status = status;
-                return status;
+                task->super.status = status;
             }
```

Comment:
need to add "return" after this line.  This return also misses from "master" branch.

Reply:
Added here and will create seperate PR for master.